### PR TITLE
Move ReactDOMLegacy implementation into RootFB

### PR DESCRIPTION
**Pull request by sebmarkbage**
Only the FB entry point has legacy mode now so we can move the remaining code in there.

Also enable disableLegacyMode in modern www builds since it doesn't expose those entry points.

Now dependent on #28709.

### CODE REVIEW

File: packages/react-dom/src/client/ReactDOMRootFB.js [hunk 1]
```diff
@@ -416,3 +457,81 @@ export function unstable_renderSubtreeIntoContainer(
     callback,
   );
 }
+
+export function unmountComponentAtNode(container: Container): boolean {
+  if (disableLegacyMode) {
```
Comment (rickhanlonii) [inline packages/react-dom/src/client/ReactDOMRootFB.js:462]: In legacy mode, can this log "has not been supported since React 18"?

**Review by rickhanlonii — APPROVED** [review]
LGTM

**Comment by gnoff** [comment]
@sebmarkbage the underlying landed

File: packages/react-dom/src/client/ReactDOMRootFB.js [hunk 2]
```diff
@@ -330,6 +339,38 @@ function legacyRenderSubtreeIntoContainer(
   return getPublicRootInstance(root);
 }
 
+export function findDOMNode(
```
Comment (gnoff) [inline packages/react-dom/src/client/ReactDOMRootFB.js:342]: Should this go somewhere else since it'll still be on secret internals for now?

Comment (sebmarkbage) [reply packages/react-dom/src/client/ReactDOMRootFB.js:343]: I added essentially a re-export of findHostInstance in the internals instead which doesn't have the extra warnings and type checks. If you're using the internals you better know what you're doing.

**Pull request merged**